tests/test_client_functional.py

```diff
@@ -2453,6 +2457,40 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url",
+    (
+        # yarl.URL.__new__ raises ValueError
+        "http://:/",
+        "http://example.org:non_int_port/",
+        # # yarl.URL.origin raises ValueError
+        "http:/",
+        "http:/example.com",
+        "http:///example.com",
+        "bluesky://d:i:d",
+        "call:+3801234567",
+        "skype:handle",
+        "slack://instance/room",
+        "steam:code",
+        "twitter://handle",
+    ),
+)
+async def test_invalid_redirect_url(
+    aiohttp_client: Any, invalid_redirect_url: Any
+) -> None:
+    headers = {hdrs.LOCATION: invalid_redirect_url}
+
+    async def handler_request(request):
+        return web.Response(status=301, headers=headers)
+
+    app = web.Application()
+    app.router.add_get("/redirect", handler_request)
+    client = await aiohttp_client(app)
+
+    with pytest.raises(InvalidRedirectURL):
```

Comment:
It is the best practice to use the `match=` regex together with `pytest.raises()`.

Reply:
sure but in some cases this error arg gets a bit modified arg e.g. http:/ is changed into http:/// - probably during some parsing/fix attempt

Reply:
This can be addressed by making a partial match, just the beginning of the string, for example. Additionally, it's possible to have another input in parametrize for exact messages, which would be useful to reuse too.

Reply:
Okay, I added partial match with additional arg

Reply:
Marking as resolved.